docs/development.adoc

```diff
@@ -354,14 +354,33 @@ $ make test-odo-login-e2e
 
 NOTE: `make test-odo-login-e2e` doesn't honour environment variable `TEST_EXEC_NODES`. So by default it runs login and logout command integration test suite on a single ginkgo test node sequentially to avoid race conditions in a parallel run.
 
-* For the entire integration test suite:
+Entire integration tests suite can be run in parallel and sequentially separately:
+
+* To run entire integration tests spec in parallel (default: 2 ginkgo test node), on a test cluster:
++
+----
+$ make test-integration-all-service-catalog
+----
++
+
+----
+$ make test-integration-all
+----
++
+
+* To run entire integration tests spec sequentially or on single ginkgo test node use enviornment variable `TEST_EXEC_NODES`:
 +
 ----
-$ make test-integration
+$ TEST_EXEC_NODES=1 make test-integration-all-service-catalog
+----
++
+
+----
+$ TEST_EXEC_NODES=1 make test-integration-all
 ----
 +
 
-NOTE: `make test-integration` doesn't honour enviornment variable `TEST_EXEC_NODES`. So by default it runs the entire integration test suite on a single ginkgo test node sequentially.
+NOTE: `make test-integration-all-service-catalog` can be successfully validated only if the cluster has service catalog enabled. Whereas `make test-integration-all` runs the test irrespective of service catalog status in the cluster.
```

Comment:
But, IIUC, `make test-integration-all`  doesn't run the tests which explicitly require service catalog to be up. Should we mention that somewhere in this line or elsewhere? Apologies if my understanding is incorrect or it has been mentioned elsewhere already.

Reply:
> Should we mention that somewhere in this line or elsewhere?

I am sorry, could not get you. 

```
Whereas `make test-integration-all` runs the test irrespective of service catalog status in the cluster.
```
This is the expiation what you are looking for i guess. Right ?